xmlRestriction.go

```diff
@@ -45,8 +46,12 @@ func (opt *Options) EndRestriction(ele xml.EndElement, protoTree []interface{})
 		opt.CurrentEle = ""
 	}
 	if !opt.Element.Empty() {
-		if !opt.ComplexType.Empty() && len(opt.ComplexType.Peek().(*ComplexType).Elements) > 0 {
-			opt.ComplexType.Peek().(*ComplexType).Elements[len(opt.ComplexType.Peek().(*ComplexType).Elements)-1] = *opt.Element.Peek().(*Element)
+		// This handles the case of a simpleContent or complexContent (XXX - is this correct?)
```

Comment:
These comments, and the rewrite of the logic, were just for my own benefit in understanding the code. The `XXX` should not be merged; I'll remove it if the comment I've added is indeed correct.